spec.md

```diff
@@ -772,6 +772,155 @@ A function or method may be referenced in a way that creates a closure out of it
+
+For example:
+
+```php
+class Example
+{
+    public string $newName = 'Me' {
+        set(string $value) {
+            if (strlen($value) < 3) {
+                throw new \Exception('Too short');
+            }
+            $this->newName = ucfirst($value);
+        }
+    }
+
+    public string $department {
+        get {
+            return $this->values[__PROPERTY__];
+        }
+        set {
+            $this->values[__PROPERTY__] = $value;
+        }
+    }
+    // or
+    public string $department {
+        get {
+            return $this->values[__PROPERTY__];
+        }
+
+        set {
+            $this->values[__PROPERTY__] = $value;
+        }
+    }
+}
+```
+
+Property hooks also support multiple short-hook variations.
+
+For a `set` hook, if the argument name and type do not need to be redefined, then they MAY be omitted.
+
+If a hook consists of a single expression, then PHP allows it to be shortened using `=>`.  In that case:
+
+* There MUST be a single space on either side of the `=>` symbol.
+* The body MUST be on the same line as the hook declaration, unless it gets prohibitively long.  If it gets prohibitively long, the developer SHOULD consider not using the short-hook syntax.
+
+```php
+class Example
+{
+    public string $myName {
+        get => __CLASS__;
+    }
+
+    public string $newName {
+        set => ucfirst($value);
+    }
+}
+```
+
+Additionally, if the following criteria are met:
+
+* There is only one hook implementation.
+* That hook uses the short-hook syntax.
+
+Then the hook MAY be listed entirely inline.  In that case,
+
+* The hook name MUST be separated from the opening brace and the arrow operator by a single space
+* The semicolon ending of the hook MUST be separated from the closing brace by a single space.
+
+For example:
+
+```php
+class Example
+{
+    public string $myName { get => __CLASS__; }
+
+    public string $newName { set => ucfirst($value); }
+}
+```
+
+Property hooks MAY also be defined in constructor-promoted properties.  However, they
+MUST be only a single hook, with a short-syntax body, defined on a single line as above.
+If those criteria are not met, then the promoted property MUST NOT have any hooks defined
+inline.
+
+```php
+class Example
+{
+    public function __construct(
+        public string $name { set => ucfirst($value); }
+    ) {}
+}
+```
+
+The following is ***not allowed*** due to the hook being too complex:
+
+```php
+class Example
+{
+    public function __construct(
+        public string $name {
+            set {
+                if (strlen($value) < 3) {
+                    throw new \Exception('Too short');
+                }
+                $this->newName = ucfirst($value);
+            }
+        }
+    ) {}
+}
+```
+
+## 4.10 Interface properties
+
+Abstract roperties may be defined in interfaces or abstract classes, but are required to
+specify if they must support `get` operations, `set` operations, or both.
```

Comment:
I've opened https://github.com/php-fig/per-coding-style/pull/111 so that lowercase `must`s aren't ambiguous. I don't think this line would actually cause any confusion, but it spurred me to submit the PR.